scripts/commit_classifier.py

```diff
@@ -355,28 +355,18 @@ def load_user(phid):
             )
 
             if self.git_repo_dir:
-                patch_proc = subprocess.Popen(
-                    ["patch", "-p1", "--no-backup-if-mismatch", "--force"],
-                    stdin=subprocess.PIPE,
-                    cwd=self.git_repo_dir,
-                )
-                patch_proc.communicate(patch.patch.encode("utf-8"))
-                assert patch_proc.returncode == 0, "Failed to apply patch"
-
-                subprocess.run(
+                fetch_proc = subprocess.run(
                     [
                         "git",
-                        "-c",
-                        f"user.name={author_name}",
-                        "-c",
-                        f"user.email={author_email}",
-                        "commit",
-                        "-am",
-                        message,
+                        "cinnabar",
+                        "fetch",
+                        f"hg::{self.git_repo_dir}",
+                        f"{revision['id']}",
```

Comment:
This is a Phabricator revision ID, it's not the hash of the commit we just applied to the local mercurial repository.

Reply:
Sorry about that. If I understand correctly, ```needed_stack``` contains the list of patches to be applied. So ```path.base_revision``` where patch is an element of needed_stack is the hash of a commit that needs to be applied?

Reply:
No, base_revision is the revision in the Mercurial repository on top of which the patch must be applied (https://github.com/mozilla/bugbug/blob/78e5b5a961f1985d7510a74de4c5741111a58d79/scripts/commit_classifier.py#L278-L282).

The process should be:
1) Get patches from Phabricator;
2) Apply each patch to the local Mercurial repository at the specified base revision;
3) Fetch the commits from the local Mercurial repository in the local git repository.

For 3, the commits come from the local Mercurial repository, Phabricator is already out of the picture.